Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Optimization.PerformanceBudgeting;
+using DCL.Optimization.Pools;
+using DCL.SDKComponents.SceneUI.Classes;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Groups;
+using DCL.SDKComponents.SceneUI.Utils;
+using ECS.Abstract;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.Common.Components;
+using ECS.StreamableLoading.Textures;
+using ECS.Unity.Textures.Components;
+using ECS.Unity.Textures.Components.Defaults;
+using SceneRunner.Scene;
+using UnityEngine;
+using Promise = ECS.StreamableLoading.Common.AssetPromise<UnityEngine.Texture2D, ECS.StreamableLoading.Textures.GetTextureIntention>;
+
+namespace DCL.SDKComponents.SceneUI.Systems.UIBackground
+{
+    [UpdateInGroup(typeof(SceneUIComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.SCENE_UI)]
+    public partial class UIBackgroundInstantiationSystem : BaseUnityLoopSystem
+    {
+        private const int ATTEMPTS_COUNT = 6;
+
+        private readonly IComponentPool<DCLImage> imagesPool;
+        private readonly ISceneData sceneData;
+        private readonly IPerformanceBudget frameTimeBudgetProvider;
+        private readonly IPerformanceBudget memoryBudgetProvider;
+
+        public UIBackgroundInstantiationSystem(
+            World world,
+            IComponentPoolsRegistry poolsRegistry,
+            ISceneData sceneData,
+            IPerformanceBudget frameTimeBudgetProvider,
+            IPerformanceBudget memoryBudgetProvider) : base(world)
+        {
+            imagesPool = poolsRegistry.GetReferenceTypePool<DCLImage>();
+            this.sceneData = sceneData;
+            this.frameTimeBudgetProvider = frameTimeBudgetProvider;
+            this.memoryBudgetProvider = memoryBudgetProvider;
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateUIBackgroundQuery(World);
+            UpdateUIBackgroundQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBUiBackground), typeof(PBUiTransform), typeof(UITransformComponent))]
```

Comment:
This `All` query is redundant as everything you need is mentioned in arguments' list